tda/data_access_test.go

```diff
@@ -0,0 +1,507 @@
+						Type:       "Person",
+						ExternalId: "HLEgiljrtoNEiyX",
+					},
+				},
+			},
+			ConsentId:      "gid:aof7m7kl966tmzct2kc75kxeisl",
+			RevokeAfterUse: true,
+		}
+
+		mockGrantConsentRequest3 := tdapb.GrantConsentRequest{
+			User: &knowledgeobjects.User{
+				User: &knowledgeobjects.User_Property_{
+					Property: &knowledgeobjects.User_Property{
+						Type: "abc",
+						Value: &objects.Value{
+							Value: &objects.Value_StringValue{StringValue: "something"},
+						},
+					},
+				},
+			},
+			ConsentId:      "gid:aof7m7kl966tmzct2kc75kxeisl",
+			RevokeAfterUse: true,
+		}
+
+		It("grants consent user id and returns response", func() {
+			mockCtrl := gomock.NewController(GinkgoT())
+			mockClient := tdamock.NewMockTrustedDataAccessAPIClient(mockCtrl)
+
+			client, err := tda.NewTestClient(mockClient)
+			Ω(err).To(Succeed())
+
+			mockClient.EXPECT().GrantConsent(
+				gomock.Any(),
+				gomock.Eq(&mockGrantConsentRequest),
+				gomock.Any(),
+			).Return(&tdapb.GrantConsentResponse{}, nil)
+
+			resp, err := client.GrantConsent(context.Background(), &mockGrantConsentRequest)
+			Ω(resp).ToNot(BeNil())
+			Ω(err).To(Succeed())
+		})
+
+		It("grants consent with external id and returns response", func() {
+			mockCtrl := gomock.NewController(GinkgoT())
+			mockClient := tdamock.NewMockTrustedDataAccessAPIClient(mockCtrl)
+
+			client, err := tda.NewTestClient(mockClient)
+			Ω(err).To(Succeed())
+
+			mockClient.EXPECT().GrantConsent(
+				gomock.Any(),
+				gomock.Eq(&mockGrantConsentRequest2),
+				gomock.Any(),
+			).Return(&tdapb.GrantConsentResponse{}, nil)
+
+			resp, err := client.GrantConsent(context.Background(), &mockGrantConsentRequest2)
+			Ω(resp).ToNot(BeNil())
+			Ω(err).To(Succeed())
+		})
+
+		It("grants consent with property and returns response", func() {
+			mockCtrl := gomock.NewController(GinkgoT())
+			mockClient := tdamock.NewMockTrustedDataAccessAPIClient(mockCtrl)
+
+			client, err := tda.NewTestClient(mockClient)
+			Ω(err).To(Succeed())
+
+			mockClient.EXPECT().GrantConsent(
+				gomock.Any(),
+				gomock.Eq(&mockGrantConsentRequest3),
+				gomock.Any(),
+			).Return(&tdapb.GrantConsentResponse{}, nil)
+
+			resp, err := client.GrantConsent(context.Background(), &mockGrantConsentRequest3)
+			Ω(resp).ToNot(BeNil())
+			Ω(err).To(Succeed())
+		})
+
+		It("handles and returns error", func() {
+			mockCtrl := gomock.NewController(GinkgoT())
+			mockClient := tdamock.NewMockTrustedDataAccessAPIClient(mockCtrl)
+
+			client, err := tda.NewTestClient(mockClient)
+			Ω(err).To(Succeed())
+
+			mockClient.EXPECT().GrantConsent(
+				gomock.Any(),
+				gomock.Eq(&mockGrantConsentRequest),
+				gomock.Any(),
+			).Return(nil, status.Error(mockErrorCode, mockErrorMessage))
+
+			resp, err := client.GrantConsent(context.Background(), &mockGrantConsentRequest)
+			Ω(resp).To(BeNil())
+			Ω(err).To(HaveOccurred())
+			Ω(err).To(test.MatchStatusError(mockErrorCode, mockErrorMessage))
+		})
+	})
```

Comment:
I would be easier to maintain and less repetitive if you use a DescribeTable.
You can do one DescribeTable by endpoints, and add a entry for each case.

Reply:
Could you be more specific because I think there is a describe table by endpoint?

Reply:
Ginko support DescribeTable tests
https://github.com/indykite/jarvis-proto/blob/40aec0cb656637fae24af692f6c5f7469bc21d1e/pkg/flow/auth/create_credentials_node_test.go#L123-L122

Reply:
ah yes!